### PR TITLE
Remote newline strip

### LINKED ISSUE #3918: Remote tests fail due to trailing newline
**Describe the bug**
The tests test_run_cmd_stdin_str and test_run_cmd_stdin_file both fail on MacOS 10.14.6 and python 3.8.6 because the output adds a trailing newline ('\n') that was not expected by the test.

**Release version(s) and/or repository branch(es) affected?**
master branch

**Steps to reproduce the bug**
Run the test suite

**Expected behavior**
All tests should pass.

**Test outpupt**
```
=================================== FAILURES ===================================
____________________________ test_run_cmd_stdin_str ____________________________
[gw0] darwin -- Python 3.8.6 /Users/jhaiduce/Development/cylc-flow/venv/bin/pyth
on3

    def test_run_cmd_stdin_str():
        """Test passing stdin as a string."""
        proc = run_cmd(
            ['sed', 's/foo/bar/'],
            stdin_str='1foo2',
            capture_process=True
        )
>       assert proc.communicate() == (
            b'1bar2',
            b''
        )
E       AssertionError: assert (b'1bar2\n', b'') == (b'1bar2', b'')
E         At index 0 diff: b'1bar2\n' != b'1bar2'
E         Full diff:
E         - (b'1bar2', b'')
E         + (b'1bar2\n', b'')
E         ?         ++

tests/unit/test_remote.py:30: AssertionError
___________________________ test_run_cmd_stdin_file ____________________________
[gw0] darwin -- Python 3.8.6 /Users/jhaiduce/Development/cylc-flow/venv/bin/python3

tmp_path = PosixPath('/private/var/folders/_c/hxz6_dsj09qf407zw_85j_rh0000gp/T/pytest-of-jhaiduce/pytest-51/popen-gw0/test_run_cmd_stdin_file0/stdin')

    def test_run_cmd_stdin_file(tmp_path):
        """Test passing stdin as a file."""
        tmp_path = tmp_path / 'stdin'
        with tmp_path.open('w+') as tmp_file:
            tmp_file.write('1foo2')
        tmp_file = tmp_path.open('rb')
        proc = run_cmd(
            ['sed', 's/foo/bar/'],
            stdin=tmp_file,
            capture_process=True
        )
>       assert proc.communicate() == (
            b'1bar2',
            b''
        )
E       AssertionError: assert (b'1bar2\n', b'') == (b'1bar2', b'')
E         At index 0 diff: b'1bar2\n' != b'1bar2'
E         Full diff:
E         - (b'1bar2', b'')
E         + (b'1bar2\n', b'')
E         ?         ++

tests/unit/test_remote.py:47: AssertionError
```

### PR DESCRIPTION
**Pull request by jhaiduce**
These changes close #3918

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] Appropriate change log entry included.
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Review by kinow — APPROVED** [review]
Looks good to me, CI passing Ok.

File: CHANGES.md [hunk 1]
```diff
@@ -95,6 +95,10 @@ config option `[scheduling]stop after cycle point`.
 
 ### Fixes
 
+[#3919](https://github.com/cylc/cylc-flow/pull/3919) - Fixed a bug in which
+remote tests failed when a newline character was included in the output from
+sed.
+
```
Comment (MetRonnie) [inline CHANGES.md:101]: Probably best to drop the changelog commit as this is just a test fix

Comment (jhaiduce) [reply CHANGES.md:101]: Fine with me. I'll remove it.

**Review by hjoliver — APPROVED** [review]
Thanks @jhaiduce :+1:

**Pull request merged**